### PR TITLE
Fix #23055 Change pd.to_datetime and pd.to_timedelta to always convert null-like argument into pd.NaT

### LINKED ISSUE #23055: Converting None values with pandas.to_datetime is unpredictable
Why does ``pandas`` convert ``None`` values in the different ways for ``to_datetime`` (unpredictable) and ``to_numeric`` (predictable)?

```python
import pandas as pd
  
VALUE = None
    
print(pd.to_datetime(VALUE))
print(pd.to_numeric(VALUE))

print(pd.__version__)
```

returns

```
None
nan
0.23.4
```

due to https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/tools/datetimes.py#L382

Why not ``pd.to_datetime(None) is pd.NaT`` by design?

See also [original post at stackoverflow](https://stackoverflow.com/questions/52720322/converting-none-values-with-pandas-to-datetime-is-unpredictable).

### PR DESCRIPTION
**Pull request by ghost**
I've tried to address the bug in `to_datetime` and `to_timedelta` functions. Please let me know if there're some other places worth to be changed.

- [x] closes #23055
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9333,8 +9333,13 @@ def truncate(
         if ax.is_all_dates:
             from pandas.core.tools.datetimes import to_datetime
 
-            before = to_datetime(before)
-            after = to_datetime(after)
+            # We do not convert None to NaT
+            # since None means means absent bound
```
Comment (jreback) [inline pandas/core/generic.py:9337]: does this actually break any tests?

File: pandas/tests/tools/test_to_datetime.py [hunk 2]
```diff
@@ -2402,3 +2402,8 @@ def test_na_to_datetime(nulls_fixture, klass):
     result = pd.to_datetime(klass([nulls_fixture]))
 
     assert result[0] is pd.NaT
+
+
+def test_none_to_datetime(nulls_fixture):
+    result = pd.to_datetime(nulls_fixture)
```
Comment (jreback) [inline pandas/tests/tools/test_to_datetime.py:2408]: can you also do a list of the fixture

File: pandas/tests/tools/test_to_timedelta.py [hunk 3]
```diff
@@ -155,3 +155,7 @@ def test_to_timedelta_float(self):
         result = pd.to_timedelta(arr, unit="s")
         expected_asi8 = np.arange(999990000, int(1e9), 1000, dtype="int64")
         tm.assert_numpy_array_equal(result.asi8, expected_asi8)
+
+    def test_none_to_timedelta(self, nulls_fixture):
+        result = pd.to_timedelta(nulls_fixture)
```
Comment (jreback) [inline pandas/tests/tools/test_to_timedelta.py:160]: same

add the issue number as a comment

File: pandas/core/tools/timedeltas.py [hunk 4]
```diff
@@ -87,8 +88,8 @@ def to_timedelta(arg, unit="ns", errors="raise"):
             "represent unambiguous timedelta values durations."
         )
 
-    if arg is None:
-        return arg
+    if arg is None or arg is libmissing.NA:
+        return NaT
```
Comment (jreback) [inline pandas/core/tools/timedeltas.py:92]: we are testing NA ?

File: doc/source/whatsnew/v1.1.0.rst [hunk 5]
```diff
@@ -687,6 +687,7 @@ Conversion
 - Bug in :class:`Series` construction from NumPy array with big-endian ``datetime64`` dtype (:issue:`29684`)
 - Bug in :class:`Timedelta` construction with large nanoseconds keyword value (:issue:`32402`)
 - Bug in :class:`DataFrame` construction where sets would be duplicated rather than raising (:issue:`32582`)
+- Inconsistency in :func:`to_datetime` and :func:`to_timedelta` when converting null-like values (:issue:`23055`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:690]: this is not very descriptive, this is *only* for ``None`` where we don't convert

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you also add tests for null-likes in Timestamp / Timdelta as well

**Comment by simonjayhawkins** [comment]
@dair-targ are you still working on this? can you address comments. ping if you need help.

**Comment by simonjayhawkins** [comment]
closing as stale. ping to continue or open a new PR.
